python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -304,13 +304,30 @@ def _start_worker_group(
 
         # Check for `bundle_label_selector` to influence WorkerGroup scheduling.
         bundle_label_selector = None
+        if isinstance(scaling_config.bundle_label_selector, list):
+            bundle_label_selector = scaling_config.bundle_label_selector[:num_workers]
```

Comment:
assert len() == num_workers. Why take the first [:num_workers]?

Reply:
discussed offline